### ISSUE TITLE
DC does not handle NewPipe email intents properly

### ISSUE THREAD

**Issue by errhammr** [deltachat/deltachat-android#1701]
- Operating System (Linux/Mac/Windows/iOS/Android): Android
- Delta Chat Version: 1.12.5
- Expected behavior: DC should open a chat with the email address that was passed in the intent and pre-fill the text field with the email body that was passed.
- Actual behavior: DC opens the chat list, like it does when it is opened using the app icon in the launcher.
- Steps to reproduce the problem: Use an app that can send email intents (i.e. a "report an error using email") to send an email intent to DC. (more info below, this is most likely **not** a problem caused by another app)
- Screenshots:
- Logs:

Let me tell you how I discovered this issue.

I tried to watch a YouTube video using the NewPipe app (version 0.20.1 from F-Droid). The app failed to load the video and showed me an error message. At the very bottom of the error screen there is a button labelled "report error via email". So I pressed that button and my Android asked me which email client I'd like to use to send the email. I picked DC from the list of installed email clients. Then DC opened and showed me the chat list, which was unexpected behaviour. Instead, I expected DC to open a chat to the email address that was passed to DC via the intent and fill the email text into the text field.

The story continues. Since I wanted to know what was going on, I repeated what I did previously. This time, when my Android asked me which email client to use, I chose FairEmail instead of DC. FairEmail opened the "write an email" screen and filled in the recipient, the subject and the email body using the corresponding data fields of the email intent.

Then I tried triggering an email intent from a web browser (Fennec version 82.1.1 from F-Droid). I created [this gist](https://gist.github.com/errhammr/afe737082c95850fb63a4c36d2e3ff33) to test different `mailto:` URIs. Both DC and FairEmail behaved as expected, filling out the recipient address, subject and body correctly.

To make sure this is **not** an issue **caused by NewPipe**, I then tried a different app to trigger an email intent. I used the "report a problem" feature of the App Unstoppable Wallet (version 0.16.2 from F-Droid). This email intent only contains the recipient email address, no subject and no body. As with NewPipe, DC did not behave as expected and only showed me the chat list instead of opening a chat to the email address that was passed to DC via the intent. I repeated the steps with FairEmail and FairEmail did open the "write an email" screen with the recipient filled out correctly.

As far as I can tell, this is most likely a problem caused by DC because FairEmail seems to handle the very same email intents correctly.

**Comment by errhammr** [comment]
Update: I've found an app that sends email intents that are handled partially correctly by DC. It opens a chat with the recipient email address but ignores the subject passed via the intent. The intent does not include a email body.
Again, FairEmail handles this correctly by filling out the recipient and the subject line.

The app is:
Contact Diary - https://f-droid.org/packages/com.apozas.contactdiary

**Comment by gerryfrancis** [comment]
@errhammr Thank you for your report, I have installed NewPipe and could not watch a video, too, so it was easy to confirm the issue. ;) I have installed ContactDiary, too. The difference I can see between these two apps is that NewPipe, when crashed, adds a body that contains the crash data, while ContactDiary does not add a body when you just want to contact the developers. It seems that Delta Chat cannot handle an extremely long mailto link that is passed to it, or the body generated by NewPipe contains something Delta Chat does not "like"...
